### DISCUSSION TITLE
code samples needed

### DISCUSSION THREAD

**Discussion by figuerres** [pnp/pnpframework#484] [Q&A]
i am trying to get started and can't seem to find much good example code to help me out.

when i  do get context it seems like something is wrong as many properties show not initialized errors but i am not sure if that is normal or not...
 
i need to upload files to a document library.....  but i get file not found errors no matter what i do.
i cant tell if i never got connected to the SharePoint site or what....

**Comment by NikoMix** [comment]
Hello,

[Here you can find some examples how to upload files to a document library.](https://github.com/pnp/pnpframework/blob/e72165aa29a7eccf9607c12b05cef7cde7e06bd4/src/lib/PnP.Framework.Test/Framework/Connectors/ConnectorSharePointTests.cs#L23)

[For setting up your context here you find a little more information.](https://github.com/pnp/pnpframework/blob/e72165aa29a7eccf9607c12b05cef7cde7e06bd4/src/lib/PnP.Framework.Test/Framework/Connectors/ConnectorSharePointTests.cs#L23)

If you can share a sample project or similar, I might be able to help more.

Please note: I'm not representing my company in this matter, just myself.

**Comment by jansenbe** [comment]
Alternatively use PnP Core SDK (https://github.com/pnp/pnpcore/tree/dev) which is our new SDK. That one has an extensive documentation, samples and tutorials: https://pnp.github.io/pnpcore/using-the-sdk/readme.html.

Not all features (the provisioning engine is the biggest pending task) are already ported to the new SDK, so you might not be able to only use PnP Core SDK. There's however interop with PnP Framework (see https://pnp.github.io/pnpcore/using-the-sdk/basics-pnpframework.html#using-pnp-framework-when-the-pnp-core-sdk-was-already-configured)
